src/styles/globals.scss

```diff
@@ -9,6 +9,12 @@ body {
   scroll-padding-top: 4.25rem;
 }
 
+/* stylelint-disable-next-line selector-no-qualifying-type */
+body:has(dialog[open]) {
+  overflow: hidden;
+  position: fixed;
```

Comment:
this causes a shift when the dialog opens (the scroll position gets sent back to the top)

I think we could just put off fixing scrolling behind dialogs and blame it on browsers instead. Maybe it'll be fixed in a few years (https://github.com/whatwg/html/issues/7732)